File: app/src/mock/java/com/luigi573/mira/data/source/remote/FakeRemoteDataSource.java
package com.luigi573.mira.data.source.remote;

import com.luigi573.mira.data.Statistics;
import com.luigi573.mira.data.source.Remote;
import com.luigi573.mira.data.source.StatisticsDataSource;

import javax.inject.Inject;
import javax.inject.Singleton;

import io.reactivex.Flowable;

@Singleton
@Remote
public class FakeRemoteDataSource implements StatisticsDataSource {
    @Inject
    public FakeRemoteDataSource() {
    }


    @Override
    public Flowable<Statistics> getStatistics() {
        return Flowable.just(new Statistics(111
                ,111
                ,111
                ,111
                ,111
                , "https://img1.dxycdn.com/2020/0123/733/3392575782185696736-73.jpg"));
    }

    @Override
    public Flowable<Statistics> getStatistics(String area) {
        return null;
    }
}

File: app/src/mock/java/com/luigi573/mira/data/source/remote/FakeRemoteDataSource.kt
package com.luigi573.mira.data.source.remote

import com.luigi573.mira.data.Result.Success
import com.luigi573.mira.data.Stat
import com.luigi573.mira.data.source.StatDataSource

class FakeRemoteDataSource : StatDataSource {
    override suspend fun getStat() = Success(Stat(
            111,
            111,
            111,
            111,
            1,
            "https://img1.dxycdn.com/2020/0123/733/3392575782185696736-73.jpg"))
}